pkg/config/config_test.go

```diff
@@ -0,0 +1,104 @@
+/*
+ Copyright 2025, NVIDIA CORPORATION & AFFILIATES
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package config_test
+
+import (
+	"github.com/Mellanox/spectrum-x-operator/pkg/config"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Test Config via API", func() {
+
+	Context("Parse Config valid input", func() {
+		It("Should return Config", func() {
+			cfg, err := config.ParseConfig(validConfig())
+			Expect(err).ToNot(HaveOccurred())
+			Expect(cfg).To(Not(BeNil()))
```

Comment:
to be sure that nothing is breaking you can marshal the output struct and compare the output, something like 

```go
	out, err := json.Marshal(cfg)
	Expect(err).ToNot(HaveOccurred())
	data = jsonData
	data = strings.ReplaceAll(data, " ", "")
	data = strings.ReplaceAll(data, "\n", "")
	data = strings.ReplaceAll(data, "\t", "")
	Expect(out).To(Equal(data))
```

there are probably more elegant ways to compare the output

Reply:
Done